pkg/policy/verifier.go

```diff
@@ -0,0 +1,56 @@
+/*
+Copyright 2022 Google LLC All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package policy
+
+import (
+	"fmt"
+
+	"github.com/google/go-containerregistry/pkg/name"
+	"knative.dev/pkg/apis"
+)
+
+// Verifier is the interface for checking that a given image digest satisfies
+// the policies backing this interface.
+type Verifier interface {
+	// Verify checks that the provided digest satisfies the backing policies.
+	Verify(name.Digest) error
+}
+
+// Compile turns a Verification into an executable Verifier.
+// Any compilation errors are returned here.
+func Compile(v Verification) (Verifier, error) {
+	return &impl{
+		verification: v,
+	}, nil
+}
+
+type impl struct {
+	verification Verification
+}
+
+// Check that impl implements Verifier
+var _ Verifier = (*impl)(nil)
+
+// Verify implements Verifier
+func (i *impl) Verify(d name.Digest) error {
+	err := &apis.FieldError{
+		Message: "NYI",
+		Details: fmt.Sprintf("got policies: %v", i.verification),
+	}
+
+	return err.At(apis.WarningLevel)
```

Comment:
This should basically be a call to `ValidatePolicy` with any policies that match the digest.